File: ts/src/common/forms/directives/ng_control_status.ts
import {Directive} from 'angular2/src/core/metadata';
import {Self} from 'angular2/src/core/di';
import {NgControl} from './ng_control';
import {isBlank, isPresent} from 'angular2/src/facade/lang';

@Directive({
  selector: '[ng-control],[ng-model],[ng-form-control]',
  host: {
    '[class.ng-untouched]': 'ngClassUntouched',
    '[class.ng-touched]': 'ngClassTouched',
    '[class.ng-pristine]': 'ngClassPristine',
    '[class.ng-dirty]': 'ngClassDirty',
    '[class.ng-valid]': 'ngClassValid',
    '[class.ng-invalid]': 'ngClassInvalid'
  }
})
export class NgControlStatus {
  private _cd: NgControl;

  constructor(@Self() cd: NgControl) { this._cd = cd; }

  get ngClassUntouched(): boolean {
    return isPresent(this._cd.control) ? this._cd.control.untouched : false;
  }
  get ngClassTouched(): boolean {
    return isPresent(this._cd.control) ? this._cd.control.touched : false;
  }
  get ngClassPristine(): boolean {
    return isPresent(this._cd.control) ? this._cd.control.pristine : false;
  }
  get ngClassDirty(): boolean {
    return isPresent(this._cd.control) ? this._cd.control.dirty : false;
  }
  get ngClassValid(): boolean {
    return isPresent(this._cd.control) ? this._cd.control.valid : false;
  }
  get ngClassInvalid(): boolean {
    return isPresent(this._cd.control) ? !this._cd.control.valid : false;
  }
}

File: lib/src/common/forms/directives/ng_control_status.dart
library angular2.src.common.forms.directives.ng_control_status;

import "package:angular2/core.dart" show Directive, Self;
import "ng_control.dart" show NgControl;
import "package:angular2/src/facade/lang.dart" show isBlank, isPresent;

/**
 * Directive automatically applied to Angular forms that sets CSS classes
 * based on control status (valid/invalid/dirty/etc).
 */
@Directive(
    selector: "[ngControl],[ngModel],[ngFormControl]",
    host: const {
      "[class.ng-untouched]": "ngClassUntouched",
      "[class.ng-touched]": "ngClassTouched",
      "[class.ng-pristine]": "ngClassPristine",
      "[class.ng-dirty]": "ngClassDirty",
      "[class.ng-valid]": "ngClassValid",
      "[class.ng-invalid]": "ngClassInvalid"
    })
class NgControlStatus {
  NgControl _cd;
  NgControlStatus(@Self() NgControl cd) {
    this._cd = cd;
  }
  bool get ngClassUntouched {
    return isPresent(this._cd.control) ? this._cd.control.untouched : false;
  }

  bool get ngClassTouched {
    return isPresent(this._cd.control) ? this._cd.control.touched : false;
  }

  bool get ngClassPristine {
    return isPresent(this._cd.control) ? this._cd.control.pristine : false;
  }

  bool get ngClassDirty {
    return isPresent(this._cd.control) ? this._cd.control.dirty : false;
  }

  bool get ngClassValid {
    return isPresent(this._cd.control) ? this._cd.control.valid : false;
  }

  bool get ngClassInvalid {
    return isPresent(this._cd.control) ? !this._cd.control.valid : false;
  }
}
